inspirehep/records/serializers/bibtex.py

```diff
@@ -0,0 +1,168 @@
+        "url",
+        "year",
+    ]
+
+    FIELDS_FOR_ENTRY_TYPE = {
+        "techreport": ["author", "number", "address", "type", "institution"],
+        "phdthesis": ["reportNumber", "school", "address", "type", "author"],
+        "inproceedings": [
+            "publisher",
+            "author",
+            "series",
+            "booktitle",
+            "number",
+            "volume",
+            "reportNumber",
+            "editor",
+            "address",
+            "organization",
+            "pages",
+        ],
+        "misc": ["howpublished", "reportNumber", "author"],
+        "mastersthesis": ["reportNumber", "school", "address", "type", "author"],
+        "proceedings": [
+            "publisher",
+            "series",
+            "number",
+            "volume",
+            "reportNumber",
+            "editor",
+            "address",
+            "organization",
+            "pages",
+        ],
+        "book": [
+            "publisher",
+            "isbn",
+            "author",
+            "series",
+            "number",
+            "volume",
+            "edition",
+            "editor",
+            "reportNumber",
+            "address",
+        ],
+        "inbook": [
+            "chapter",
+            "publisher",
+            "author",
+            "series",
+            "number",
+            "volume",
+            "edition",
+            "editor",
+            "reportNumber",
+            "address",
+            "type",
+            "pages",
+        ],
+        "article": ["author", "journal", "number", "volume", "reportNumber", "pages"],
+    }
+
+    def __init__(self, schema_class=BibTexCommonSchema):
+        self.schema_class = schema_class()
+
+    def create_bibliography_entry(self, record):
+        bibtex_document_type = self.schema_class.get_bibtex_document_type(record)
+
+        data = self.schema_class.dump(record).data
+        doc_type = data.pop("doc_type", None)
+        texkey = data.pop("texkey", None)
+        fields = (
+            self.COMMON_FIELDS_FOR_ENTRIES
+            + self.FIELDS_FOR_ENTRY_TYPE[bibtex_document_type]
+        )
+        template_data = [
+            (key, str(value)) for key, value in data.items() if value and key in fields
+        ]
+        data_bibtex = [
+            texkey,
+            Entry(
+                doc_type,
+                template_data,
+                persons={
+                    "author": [
+                        Person(x)
+                        for x in BibTexCommonSchema.get_authors_with_role(
+                            data.get("authors", []), "author"
+                        )
+                    ],
+                    "editor": [
+                        Person(x)
+                        for x in BibTexCommonSchema.get_authors_with_role(
+                            data.get("authors", []), "editor"
+                        )
+                    ],
+                },
+            ),
+        ]
```

Comment:
This huge statement is not very readable..
I would prefer creating `author` and `editor` out of the `person` kwarg, at least